mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -719,3 +720,40 @@ def get_bad_information(eeg, get_pos, *, montage_units):
     assert len(pos["lpa"]) == 3
     assert len(pos["rpa"]) == 3
     assert len(raw.info["dig"]) == n_eeg + 3
+
+
+@testing.requires_testing_data
+def test_eeglab_drop_nan_annotations(tmp_path):
+    """Test reading file with NaN annotations."""
+    pytest.importorskip("eeglabio")
+    file_path = tmp_path / "test_nan_anno.set"
+    raw = read_raw_eeglab(raw_fname_mat, preload=True)
+    data = raw.get_data()
+    sfreq = raw.info["sfreq"]
+    ch_names = raw.ch_names
+    anno = [
+        raw.annotations.description,
+        raw.annotations.onset,
+        raw.annotations.duration,
+    ]
+    anno[1][0] = np.nan
+
+    export_set(
+        str(file_path),
+        data,
+        sfreq,
+        ch_names,
+        ch_locs=None,
+        annotations=anno,
+        ref_channels="common",
+        ch_types=np.repeat("EEG", len(ch_names)),
+    )
+
+    with pytest.raises(
+        RuntimeWarning,
+        match=(
+            r"1 events have an onset that is NaN. These values are usually "
+            r"ignored by EEGLAB and will be dropped from the annotations."
+        ),
+    ):
```

Comment:
Usually it's best to make these as short as possible while still being selective, that way if we tweak wording later in the warning for example we don't need to update tests
```suggestion
    with pytest.raises(RuntimeWarning, match=r"1 events have an onset that is NaN.*"):
```